compiler/rustc_errors/src/diagnostic.rs

```diff
@@ -394,16 +396,10 @@ impl Diagnostic {
         };
         let mut msg: Vec<_> =
             vec![(format!("{}{} `", " ".repeat(expected_padding), expected_label), Style::NoStyle)];
-        msg.extend(expected.0.iter().map(|x| match *x {
-            StringPart::Normal(ref s) => (s.to_owned(), Style::NoStyle),
-            StringPart::Highlighted(ref s) => (s.to_owned(), Style::Highlight),
-        }));
+        msg.extend(expected.0.into_iter().map(|p| (p.content, p.style)));
```

Comment:
Could keep using the struct and use its constructors in the handwritten cases

Reply:
Sorry, I don't understand. Can you clarify?

Reply:
Basically: make `highlighted_note` take a `Vec<StringPart>`. Then turn all of the tuple constructors to `StringPart::normal()` or `StringPart::highlighted()` calls

Reply:
Done.